### ISSUE TITLE
MAINTAINER & CONTRIBUTOR NEEDED

### ISSUE THREAD

**Issue by ohld** [ohld/igbot#1001]
Hi guys!

Instabot family needs **maintainers and contributors**. We need python experts to help our community of 5000+ users solve issues and fix bugs. I really want to expand our family and invite YOU to join our open-source organization.

I've started this project 4 years ago as my student course project and now it becomes really **HUGE**. It is so hard to do this alone so I really appreciate any help with:

1. Bug fixings
2. Issue solving
3. New Features creation
4. Documentation enhancing 

You don't need my permission to start contributing! Just create pull requests with your changes and I will check them. 

See you!

**Comment by disayner** [comment]
Unfortunately, I'm not a python expert yet. But I just wanted to say thanks for this project;) If possible I will help as much as I can.

**Comment by ohld** [comment]
@disayner thanks for your help! For all ways how you can contribute to the project, visit this page:

https://github.com/instagrambot/instabot/blob/master/CONTRIBUTING.md

**Comment by bruvv** [comment]
Perhaps let's star with clearing the issue tab. And writing a good wiki for people to understand ? And use a bot that automatically closes an issue when people do not fill out the issue template. Hate it when they do not do that.

**Comment by AKwoKWH** [comment]
> Perhaps let's star with clearing the issue tab. And writing a good wiki for people to understand ? And use a bot that automatically closes an issue when people do not fill out the issue template. Hate it when they do not do that.

Agree. The Issues Tab is now too many post just saying something like: "I cant bah bah bah, can you help?" We should clean up these posts so that we can focus on those real issues related to coding.

Also @ohld , I suggest to keep log on each request and response so whenever something broken, especially those related to the instagram api, it is much more easier for us to monitor and traceback.
(i.e people can even provide a more detailed log when they ave issues)

I have implemented these request sent record in my own repo for my debug.
I can make a push here.

```
#BEFORE REQUEST
            self.logger.debug('Request Endpoint: {}'.format(str(APIURL + endpoint)))
            self.logger.debug('Request Data: {}'.format(str(post)[0:1000]))

#AFTER REQUEST          
            self.logger.debug('Response Data: {}'.format(str(response)[0:1000]))
```

It is **important** to have these information when dealing with instagram API.
You need to know what instagram server response (i.e if you are being blocked, or you sent an incorrect endpoint, or you included an incorrect data, or you any **NEW** type of challenge required message, or Instagram return a different format (usually **None response**) causing the code below failed)

And this should also be included in the issues template, else it is hard to help.

**Comment by fighterii** [comment]
I also like to contribute. It would be great, if some contributors have the right to label issues, so we could sort out which issues really are about bugs / features and which are from people not familiar with programming/using python.

@AKwoKWH : haven't looked into it in detail, but maybe there is a good spot to log this detailed requests only in case of failure (= non 200 response code)

**Comment by ohld** [comment]
@fighterii I can grant you some rights to label issues (I hope that this is possible) if you want to help with it. This truly can help with sorting things out.

**Comment by Amanda888** [comment]
@[
Duplicate of #](url)

**Comment by leon-wbr** [comment]
I'll try my best to contribute as this seems to be the only active library supporting photo upload. I was thinking it would probably be a good idea to start writing some documentation in order to give an overview for future contributors (or even just auto generate it).

**Comment by bruvv** [comment]
That will be great! Thanks
